Documentation/BasicOperators.md

```diff
@@ -46,41 +46,362 @@ code-style will be used.
+            println("Started")
+        }, event: { event in
+            println("Event: \(event)")
+        }, error: { error in
+            println("Error: \(error)")
+        }, completed: {
+            println("Completed")
+        }, interrupted: {
+            println("Interrupted")
+        }, terminated: {
+            println("Terminated")
+        }, disposed: {
+            println("Disposed")
+        }, next: { next in
+            println("Next: \(next)")
+        })
+```
 
 ## Operator composition
 
-### Lifting
 ### Pipe
 
+The `|>` operator can be used to apply a signal operator to a signal. Multiple operators can be chained after eachother using the `|>` operator
+
+```Swift
+intSignal
+    |> filter { num in num % 2 == 0 }
+    |> map(toString)
+    |> observe(next: { string in println(string) })
+```
+
+### Lifting
+
+Signal operators can be _lifted_ to operate upon `SignalProducer`s instead using the `lift` operator.
+In other words, this will create a new `SignalProducer` which will apply the given signal operator to _every_ signal created from the incoming `SignalProducer`s
+just if the operator had been applied to each signal yielded from `start()`.
+
+The `|>` operator implicitely lifts signal operators, when used with `SignalProducer`s.
 
 ## Transforming signals
 
+These operators transform a signal into a new signal.
+
 ### Mapping
+
+The `map` operator is used to transform the values in a signal, creating a new signal with the results.
+
+```Swift
+let (signal, sink) = Signal<String, NoError>.pipe()
+signal
+    |> map { string in string.uppercaseString }
+    |> observe(next: { println($0) })
+
+sendNext(sink, "a")     // Prints A
+sendNext(sink, "b")     // Prints B
+sendNext(sink, "c")     // Prints C
+```
+
+
 ### Filtering
+
+The `filter` operator is used to include only values in a signal that satisfy a predicate
+
+```Swift
+let (signal, sink) = Signal<Int, NoError>.pipe()
+signal
+    |> filter { number in number % 2 == 0 }
+    |> observe(next: { println($0) })
+
+sendNext(sink, 1)     // Not printed
+sendNext(sink, 2)     // Prints 2
+sendNext(sink, 3)     // Not printed
+sendNext(sink, 4)     // prints 4
+```
+
 ### Reducing
 
+The `reduce` operator is used to aggregate a signals value into a signle combine value. Note, that the final value is only sended after the source signal completes.
+
+```Swift
+let (signal, sink) = Signal<Int, NoError>.pipe()
+
+signal
+    |> reduce(1) { $0 * $1 }
+    |> observe(next: { println($0) })
+
+sendNext(sink, 1)     // nothing printed
+sendNext(sink, 2)     // nothing printed
+sendNext(sink, 3)     // nothing printed
+sendCompleted(sink)   // prints 6
+```
+
 
 ## Combining signals
 
+These operators combine multiple signals into a single new signal.
+
 ### Combining latest values
+
+The `combineLatest` function combines the latest values of two (or more) signals. The resulting signal will not send a value until both inputs have sent at least one value each.
```

Comment:
Ok, I've tried to change the second sentence to
> The resulting signal will only send a the first value after both inputs have sent at least one value each. After that, each value on either of the inputs will cause a new value on the output.

And also added a counterpart to `zip`:

> The elements of any Nth pair are the Nth elements of the two input signals. That means the output signal will always wait for both input signals to send and output.